templates/core/about/builds.html

```diff
@@ -7,72 +7,76 @@
+		<code>Cargo.toml</code>. Omitting this field means that no README will be displayed.
 	</p>
 
 	<h4>Detecting Docs.rs from <code>build.rs</code></h4>
 	<p>
-		Docs.rs builds crates with the environment variable <code>DOCS_RS</code> set to
-		<code>1</code>, which enables the crate to detect docs.rs and build the
-		crate differently. This can be helpful if you need
+		Docs.rs sets the environment variable <code>DOCS_RS</code> to <code>1</code>.
+		Detecting Doc.rs and building your crate differently can be helpful if you need
 		dependencies for building the library, but not for building the documentation.
 	</p>
 
 	<h4>Detecting Docs.rs from <code>#[cfg]</code> attributes</h4>
 	<p>
-		You can detect Docs.rs by having a <a href="https://doc.rust-lang.org/cargo/reference/features.html">feature</a>
-		which is only set by Docs.rs. See <a href="metadata">Metadata</a> for more information.
+		Docs.rs sets a <a href="https://doc.rust-lang.org/cargo/reference/features.html">feature</a>. See <a href="metadata">Metadata</a> for details.
+	</p>
+
+	<h4>Testing documentation builds locally</h4>
+	{%- set build_subcommand = docsrs_repo ~ "/blob/master/README.md#build-subcommand" -%}
+	<p>
+		The Doc.rs <a href="{{ build_subcommand | safe }}">README</a> describes how to build
+		unpublished crate documentation locally using the same build environment as the Doc.rs build agent.
 	</p>
 
-	<h4>Global sandbox limits</h4>
+	<h3>Diagnosing failed builds</h3>
 
+	<h4>Missing dependencies</h4>
 	<p>
-		All the builds on docs.rs are executed inside a sandbox with limited
-		resources. The current limits are the following:
+		Missing dependencies are a common reason for failed builds.
+		Docs.rs dependencies are managed through
+		<a href="https://github.com/rust-lang/crates-build-env">crates-build-env</a>.
+		See <a href="https://forge.rust-lang.org/docs-rs/add-dependencies.html">Forge</a> for instructions
+		on how to make a PR. You can always <a href="{{ docsrs_repo | safe }}/issues">file an issue</a>
+		when having trouble.
 	</p>
 
-	{{ macros::crate_limits(limits=limits) }}
+	<h4>Writing files to read-only directories</h4>
+	<p>
+		Most of the sandbox is a read-only file system, including the source directory of your crate and its dependencies.
+		If you <code>build.rs</code> generates files that are relevant for documentation, consider writing to the output directory which is passed as the environment variable <code>OUT_DIR</code>.
+	</p>
 
+	<h4>Hitting sandbox limits</h4>
 	<p>
-		If a build fails because it hit one of those limits please
+		All the builds on Docs.rs are executed inside a sandbox with limited
+		resources. If a build fails because it hit limits please
 		<a href="{{ docsrs_repo | safe }}/issues/new/choose">open an issue</a>
 		to get them increased for your crate.
-		Note that network access will not be enabled for any crate.
+		Network access will not be enabled for any crate.
+		The current limits are:
 	</p>
 
```

Comment:
Could you add a "Requesting Rebuilds" section that outlines how to request a rebuild and why you can/can't?
You can request a rebuild by opening an issue on our repo, but we can rebuild crates that fail due to something that is *our* fault, like a missing dependency or docs.rs bug. Crates that hit resource limits *might* be rebuilt if we have the resources to do so (since our builder itself has finite resources), but failures due to networking will not be rebuilt

Reply:
And if it needs a change to the source code a rebuild won't help (e.g. a dependency was fixed, but in a semver breaking bump).

Reply:
Good idea. Will do so.